lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -224,17 +224,24 @@ namespace Js
         return JavascriptOperators::GetTypeId(aValue) == TypeIds_String;
     }
 
-    Js::PropertyRecord const * JavascriptString::GetPropertyRecord(bool dontLookupFromDictionary)
+    void JavascriptString::GetPropertyRecord(_Out_ Js::PropertyRecord const ** propertyRecord, bool dontLookupFromDictionary)
     {
+        *propertyRecord = nullptr;
         if (dontLookupFromDictionary)
         {
-            return nullptr;
+            return;
         }
 
-        Js::PropertyRecord const * propertyRecord;
-        GetScriptContext()->GetOrAddPropertyRecord(GetString(), GetLength(), &propertyRecord);
+        GetScriptContext()->GetOrAddPropertyRecord(GetString(), GetLength(), propertyRecord);
+    }
+
+    Js::PropertyId JavascriptString::GetPropertyId()
+    {
+        Js::PropertyRecord const * propertyRecord = nullptr;
+        GetPropertyRecord(&propertyRecord);
 
-        return propertyRecord;
+        Assert(propertyRecord != nullptr);
+        return propertyRecord->GetPropertyId();
```

Comment:
The propertyRecord could still be collected after this GetPropertyId() and propertyId becomes invalid, right?

Reply:
For PropertyString?